### ISSUE TITLE
GCop658 false alarm

### ISSUE THREAD

**Issue by sungam3r** [Geeksltd/GCop#290]
Before fix:

```c#
 public class NumberSizePageFilter
    {
        public int PageNumber { get; set; }
        public int PageSize { get; set; }
        public NumberSizePageFilter Next() => new NumberSizePageFilter { PageNumber = PageNumber + 1, PageSize = PageSize }; // GCop658	Use compound assignment.	
    }
```

After fix:

```c#
 public class NumberSizePageFilter
    {
        public int PageNumber { get; set; }
        public int PageSize { get; set; }
        public NumberSizePageFilter Next() => new NumberSizePageFilter { PageNumber += 1, PageSize = PageSize }; // CS0747	Invalid initializer member declarator
    }
```
which is obviously not compiled.

**Comment by sungam3r** [comment]
Any progress on this?
